src/App.tsx

```diff
@@ -1,57 +1,113 @@
-import React from 'react';
+import React, { useEffect, useLayoutEffect } from 'react';
 import Button from '@material-ui/core/Button';
 import AppBar from '@material-ui/core/AppBar';
 import Toolbar from '@material-ui/core/Toolbar';
-import { useSelector, useDispatch, useStore } from 'react-redux';
+import { useSelector, useDispatch } from 'react-redux';
 import { selectTheme, selectLanguage } from './actions/settings';
 import { useTranslation } from 'react-i18next';
-import { InitialState } from './types/default';
+import { InitialState, User } from './types/default';
 import { createMuiTheme, ThemeProvider } from '@material-ui/core/styles';
-import TestComponent from './TestComponent';
 import { LANGUAGES } from './definitions/settings';
 import { themesList } from './types/themes';
 import { THEMES } from './definitions/settings';
+import Home from './pages/Home';
+import Login from './pages/Login';
+import axios from 'axios';
+import Registration from './pages/Registration';
+import { BrowserRouter as Router, Switch, Route, Link } from 'react-router-dom';
+import { host, port, apiPrefix, entities } from './config/development.json';
+import { setUser } from './actions/user';
 
 const App: React.FC = function () {
   const dispatch = useDispatch();
   const themeState: THEMES.DARK | THEMES.DEFAULT = useSelector(
     (state: InitialState) => state.theme
   );
+  const user = useSelector((state: any) => state?.user);
   const theme = themesList[themeState];
   const createdTheme = createMuiTheme(theme);
   const { t } = useTranslation();
 
+  useLayoutEffect(() => {
+    async function getMe() {
+      if (!user?.id) {
+        const resp = await axios
+          .get(
+            host + port + apiPrefix + entities.user.selfName + entities.user.me,
+            { headers: { Authorization: localStorage.getItem('authToken') } }
```

Comment:
need to have a common solution for headers. contact me and i will share a example